src/pybind/mgr/dashboard/services/rbd.py

```diff
@@ -569,3 +571,32 @@ def snapshot_schedule_add(cls, image_spec: str, interval: str):
     @classmethod
     def snapshot_schedule_remove(cls, image_spec: str):
         _rbd_support_remote('mirror_snapshot_schedule_remove', image_spec)
+
+
+class RbdImageMetadataService(object):
+    def __init__(self, image):
+        self._image = image
+
+    def list(self):
+        result = self._image.metadata_list()
+        # filter out configuration metadata
+        return {v[0]: v[1] for v in result if not v[0].startswith('conf_')}
+
+    def get(self, name):
+        return self._image.metadata_get(name)
+
+    def set(self, name, value):
+        self._image.metadata_set(name, value)
+
+    def remove(self, name):
+        try:
+            self._image.metadata_remove(name)
+        except KeyError:
+            pass
+
+    def set_metadata(self, metadata):
+        for name, value in metadata.items():
+            if value is not None:
+                self.set(name, value)
+            else:
+                self.remove(name)
```

Comment:
Does this mean that a `None` value is equivalent to saying remove this entry? I would assume that a `name` can be set with a relevant value but if you try to set it with None, I would expect it to transform to a empty string or something like that.

If `set_metadata` should reset the metadata with a new dictionary of values we should first reset and then repopulate.

Reply:
I made it behave the same way we already have for 'configuration'.